okhttp/src/jvmMain/kotlin/okhttp3/Request.kt

```
@@ -244,7 +223,7 @@ actual class Request internal actual constructor(builder: Builder) {
     @JvmName("reifiedTag")
     actual inline fun <reified T : Any> tag(tag: T?): Builder = tag(T::class, tag)
 
-    actual fun <T : Any> tag(type: KClass<T>, tag: T?): Builder = commonTag(type, tag)
+    actual fun <T : Any> tag(type: KClass<T>, tag: T?): Builder = commonTag(type, type.cast(tag))
```

fail fast with Long/long in common code on JVM.